SAR_ADC (Chile 2)

papijou2433 opened the pull request.

papijou2433 committed:
Add files via upload

papijou2433 committed:
Delete layout/gds directory

papijou2433 committed:
SAR_ADC(Chile 1)

papijou2433 committed:
Delete SAR_ADC directory

papijou2433 committed:
SAR_ADC (Chile 2)

papijou2433 committed:
Update README.md

papijou2433 committed:
Delete SAR_ADC/gds directory

papijou2433 committed:
Add files via upload

msaligane commented:
@bmurmann Let's review the files but not merge yet. I think we will have some merge conflicts.

bmurmann reviewed (changes requested):
Some of the xschem symbols contain absolute paths, which makes the schematics non-portable and tough to review outside your personal setup (without modifying). Can you please fix all of these? Example: /foss/pdks/gf180mcuC/libs.tech/xschem/symbols/cap_mim_2p0fF.sym

papijou2433 commented:
Ok, we will be fixing the paths, thanks for reviewing the files!

d-m-bailey commented:
Just a couple housekeeping comments.
1. You probably don't need to commit `SAR_ADC/xschem/test/preamp_tb/untitled.sch`
2. `SAR_ADC/xschem/spice/strong_arm.sch` and `SAR_ADC/xschem/spice/strongarm.sch` (and the corresponding symbols) are similarly named. Are they both used and necessary? Could they be combined?

d-m-bailey commented:
Also, maybe a couple layout changes.
<img width="1213" alt="chipathon-inv" src="https://github.com/sscs-ose/DC23-LTC2/assets/8349634/5a8b488f-0874-4854-b2b4-c4c4a61a59ba">
<img width="1151" alt="chipathon-preamp" src="https://github.com/sscs-ose/DC23-LTC2/assets/8349634/18c5f0b8-94b4-4227-8a44-3151cd02555e">
<img width="1254" alt="chipathon-amp" src="https://github.com/sscs-ose/DC23-LTC2/assets/8349634/8cc9c2bd-fcc5-48fc-86d5-000e1b6e6829">

RTimothyEdwards commented:
Some suggestions:
(1) Use XSCHEM_LIBRARY_PATH in your xschemrc file to keep the full path of components out of the files:
```
     set XSCHEM_LIBRARY_PATH {}
     append XSCHEM_LIBRARY_PATH ${XSCHEM_SHAREDIR}/xschem_library
     append XSCHEM_LIBRARY_PATH :$env(PWD)
     append XSCHEM_LIBRARY_PATH :$USER_CONF_DIR/xschem_library
     append XSCHEM_LIBRARY_PATH :/foss/pdks/gf180mcuC/libs.tech/xschem
```
(2) Do not put `TT_MODELS` blocks inside the circuit schematics.  When you want to test a circuit block, create a testbench for it, instantiate the symbol of the schematic you want to test, and put the `TT_MODELS` and other simulation information into the testbench schematic, not the circuit and subcircuit schematics.
(3) Please check the PDK variant being used.  The C variant was used for the first open MPW but Global Foundries did not want us to use the thinner top metal, and so the D variant was created for all subsequent MPWs.  Regardless, both variants define the 2fF/um^2 MiM capacitor, *not* the 1.5fF/um^2 MiM capacitor.  Make sure you understand what process variant the shuttle run will be manufactured on, and use the appropriate devices accordingly.
(4) MiM capacitor dimensions have parameter names `c_length` and `c_width`, *not* `L` and `W`.  This appears to be an issue with the device symbols provided upstream.  I will try to resolve this with Stefan Schippers.  You can correct for the error in the device instances, where the parameter names are just text values.
(5) If you use 3V devices, be aware that there is no support for 3V infrastructure (e.g., padframe I/O pads)

RTimothyEdwards commented:
@msaligane :  Please read my suggestions above and comment on the PDK variant intended to be used on this tapeout, and the power supply setup expected for testing, since the PDK did not come with any 3.3V support.

RTimothyEdwards commented:
(6) Do not create multiple pins of the same name.  You may label as many wires as you want, but for each unique net that is also a pin, there should only be one pin symbol.  Otherwise, netlists generates a huge number of warnings (which may or may not be errors).

d-m-bailey commented:
> Some suggestions: (1) Use XSCHEM_LIBRARY_PATH in your xschemrc file to keep the full path of components out of the files:
> 
> ```
>      set XSCHEM_LIBRARY_PATH {}
>      append XSCHEM_LIBRARY_PATH ${XSCHEM_SHAREDIR}/xschem_library
>      append XSCHEM_LIBRARY_PATH :$env(PWD)
>      append XSCHEM_LIBRARY_PATH :$USER_CONF_DIR/xschem_library
>      append XSCHEM_LIBRARY_PATH :/foss/pdks/gf180mcuC/libs.tech/xschem
> ```
> 
Just a follow up here. Using 
```
      append XSCHEM_LIBRARY_PATH :$env(PDK_PATH)/$env(PDK)/libs.tech/xschem
```
makes your `xschemrc` file portable too.

papijou2433 commented:
@RTimothyEdwards, We weren't aware of the 3.3V issue since we were using the standard cells, does this means that we should change the 3.3V transistor sizing to be adecuate for a 4.5-5.5V transistor?

RTimothyEdwards commented:
@papijou2433 :  If you intend to use the provided demonstration/prototype board, then you probably want to make your system 5V compatible, because switching out components on the board for 3.3V operation may be difficult.

msaligane commented:
@papijou2433 In our TO the standard cells [we used are 5v](https://gf180mcu-pdk.readthedocs.io/en/latest/digital/standard_cells/gf180mcu_fd_sc_mcu7t5v0/index.html). But we used 5v and 3.3v devices : `pfet_03v3, nfet_03v3`

If I recall, in our last TO, we used: 
```
gf180mcuC   =  METALS5 | MIM | THICKMET0P9 | HRPOLY1K 
MiM: 2.0fF/um2 option B
```

@RTimothyEdwards is this what will be used in MPW1?

msaligane commented:
@papijou2433 I can merge this PR for now. But before that, please open a GitHub issue compiling all the feedback listed above or during our sync-up today. It helps with tracking things. Thanks!

RTimothyEdwards commented:
@msaligane :  It is possible to run the whole chip at 3.3V but I don't know that the existing development board is compatible with 3.3V operation (i.e., the FTDI, SPI flash, and other chips on the board).  Technically, the padframe we have from GF is 5V on the pad and core sides.  It can be run at lower voltage with a corresponding loss of speed.

msaligane commented:
@RTimothyEdwards For this TO. We are planning to use custom-made PADs. But yes, it is possible to do so.

papijou2433 committed:
Update README.md

papijou2433 committed:
Add files via upload

d-m-bailey commented:
@msaligane gfmpw-1 will use the  `gf180mcuD` process (not `gf180mcuC`). The top metal is thicker.

Commit:
Added top level simulation

Commit:
Add updated comparator, update top, add sar logic, add openlane files

Commit:
Fix test with the PDK D variant

Commit:
Add comparator testbenches

VicenteOsorio committed:
Schematic file management update

Updated file management and added new testbench files. Added a new startup testbench and makefile.

VicenteOsorio committed:
Fixed symbol paths for CDAC and TOP

Fixed symbol paths due to xschemrc

VicenteOsorio committed:
Bootstrap symbol update

Updated symbol and schematic for bootstrap switch

gsuckz committed:
bottom plate sampling

gsuckz committed:
Merge branch 'papijou2433-patch-1' of https://github.com/papijou2433/DC23-LTC2 into papijou2433-patch-1

gsuckz committed:
cambios

gsuckz committed:
Cambios

gsuckz committed:
CDAC2

gsuckz committed:
CDAC2 correcciones

gsuckz committed:
Cambios con Mario

gsuckz committed:
cambios

gsuckz committed:
Probar diferentes valores DC

gsuckz committed:
Corregir dimensiones de buff/inv

gsuckz committed:
cambios

gsuckz committed:
Agregados logic_tb

gsuckz committed:
Falta logic.spice

Commit:
Update SAR Logic

gsuckz committed:
camibios

Commit:
Fix SAR Logic sample

gsuckz committed:
Merge branch 'papijou2433-patch-1' of https://github.com/papijou2433/DC23-LTC2 into papijou2433-patch-1

gsuckz committed:
cambios

Commit:
Fix timing issue with current bit counting

Commit:
Fix timing issue with current bit counting

gsuckz committed:
Cambios con Mario

Commit:
Update RTL

Commit:
Merge branch 'papijou2433-patch-1' of https://github.com/papijou2433/DC23-LTC2 into papijou2433-patch-1

gsuckz committed:
Cambios

VicenteOsorio committed:
Updated .devcontainer and .json

Updated tools container
